server/grpc_bytestream.go

```diff
@@ -129,10 +129,9 @@ func (s *grpcServer) Read(req *bytestream.ReadRequest,
 	var chunkResp bytestream.ReadResponse
 	for {
 		n, err := rdr.Read(buf)
-		if err == io.EOF {
-			s.accessLogger.Printf("GRPC BYTESTREAM READ COMPLETED %s",
-				req.ResourceName)
-			return nil
+		eof := err == io.EOF
+		if eof {
+			err = nil
 		}
 
 		if err != nil {
```

Comment:
https://golang.org/pkg/io/#Reader says "Callers should always process the n > 0 bytes returned before considering the error err."

So how about we put an `if n > 0` block immediately after the `Read` call, with the `limitedSend` check and the `Send` call inside, and then check `if err == io.EOF { <success> }` and finally `if err != nil { <failure> }`

Reply:
Done.